pkg/apis/pipeline/v1beta1/pipeline_types.go

```diff
@@ -122,12 +122,17 @@ type PipelineResult struct {
 	// Name the given name
 	Name string `json:"name"`
 
+	// Type is the user-specified type of the result. The possible type
+	// is currently "string", "array" and "object" are alpha feature.
+	// +optional
+	Type ResultsType `json:"type,omitempty"`
+
 	// Description is a human-readable description of the result
 	// +optional
 	Description string `json:"description"`
 
 	// Value the expression used to retrieve the value
-	Value string `json:"value"`
+	Value ArrayOrString `json:"value"`
```

Comment:
could you please update the commit message/PR description noting that this PR includes a breaking change to the Go libraries?

Reply:
yes sure!